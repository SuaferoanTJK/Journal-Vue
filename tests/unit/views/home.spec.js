import { shallowMount } from "@vue/test-utils";
import Home from "@/views/HomeView";

describe("Home View Unit Testing", () => {
  test("Should render the component", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("Click in button should redirect to noEntry", () => {
    const mockRouter = {
      push: jest.fn(),
    };
    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });
    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "noEntry" });
  });
});
