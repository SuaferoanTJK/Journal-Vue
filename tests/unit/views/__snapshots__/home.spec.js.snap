// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Unit Testing Should render the component 1`] = `
<div>
  <h1>This is home page</h1><button>Go to Daybook</button>
</div>
`;
